wazo_calld/plugins/transfers/stasis.py

```diff
@@ -183,34 +196,32 @@ def hangup(self, channel, event):
         self.hangup_pubsub.publish(transfer_role, transfer)
 
     def transfer_recipient_answered(self, channel_event):
+        logger.debug('received TransferRecipientAnswered event: %s', channel_event)
         channel, event = channel_event
         event = TransferRecipientAnsweredEvent(event)
 
-        try:
-            transfer_bridge = self.ari.bridges.get(bridgeId=event.transfer_bridge)
-            transfer_bridge.addChannel(channel=channel.id)
-        except ARINotFound:
-            logger.error('recipient answered, but transfer was hung up')
-            return
-
         transfer_id = event.transfer_bridge
         try:
             with self.state_factory.make(transfer_id) as transfer_state:
                 logger.debug('recipient answered, transfer continues normally')
                 transfer_state.recipient_answer()
         except KeyError:
             logger.debug('recipient answered, but transfer was abandoned')
-
-            for channel_id in transfer_bridge.json['channels']:
-                try:
-                    ari_helpers.unring_initiator_call(self.ari, channel_id)
-                except ARINotFound:
-                    pass
+            # avoid leaving recipient channel hanging
+            channel.hangup()
```

Comment:
Why hangup? Shouldn't the initiator and recipient be bridged instead?

Reply:
The transfer doesn't exist anymore, so we can't know the initiator.

And now, as long as the recipient has not answered and the state machine did not process that event, there are no bridges. Only the state machine knows to create the bridge when the participants are ready to talk.
The refactoring also had a goal to encapsulate bridge management within the state machine and decouple other code from implementation details around bridges.

Here we can only assume that if the transfer has ended before the recipient answered, the initiator has/was hung up. If a different happening is possible where the transfer disappears but initiator channel remains, we should handle this explicitly.

Reply:
OK, works for me. In this case, the debug log above is wrong though: the transfer is not "abandoned", it was deleted, right?

Reply:
Hmm yes, fair enough it might not be an "abandon" scenario. I can change the log to be more generic.